spec/tbd_tests_spec.rb

```diff
@@ -4869,7 +5154,267 @@
+    expect(construction2.layers.size).to eq(1)
+
+    # Assign construction to the "Office Left Wall".
+    expect(left_office_wall.setConstruction(construction)).to be(true)
+
+    # Assign construction2 to the "Fine Storage Right Wall".
+    expect(right_fine_wall.setConstruction(construction2)).to be(true)
+
+    subs = front_office_wall.subSurfaces
+    expect(subs.empty?).to be(false)
+    expect(subs.size).to eq(4)
+
+    psi_set = "poor (BETBG)"
+    ioP = File.dirname(__FILE__) + "/../json/tbd_warehouse9.json"
+    # {
+    #   "schema": "https://github.com/rd2/tbd/blob/master/tbd.schema.json",
+    #   "description": "testing error detection",
+    #   "psis": [
+    #     {
+    #       "id": "detailed_2",
+    #       "fenestration": 0.600
+    #     },
+    #     {
+    #       "id": "regular (BETBG)",                    <<<< non-fatal ERROR # 1
+    #       "fenestration": 0.700
+    #     }
+    #   ],
+    #   "khis": [
+    #     {
+    #       "id": "cantilevered beam",
+    #       "point": 0.6
+    #     }
+    #   ],
+    #   "surfaces": [
+    #     {
+    #       "id": "Office Front Wall",
+    #       "khis": [
+    #         {
+    #           "id": "beam",                           <<<< non-fatal ERROR # 2
+    #           "count": 3
+    #         }
+    #       ]
+    #     },
+    #     {
+    #       "id": "Office Left Wall",
+    #       "khis": [
+    #         {
+    #           "id": "cantilevered beam",
+    #           "count": 300                           <<<< triggers WARNING # 1
+    #         }
+    #       ]
+    #     }
+    #   ],
+    #   "edges": [
+    #     {
+    #       "psi": "detailed",                          <<<< non-fatal ERROR # 3
+    #       "type": "fenestration",
+    #       "surfaces": [
+    #         "Office Front Wall",
+    #         "Office Front Wall Window 1"
+    #       ]
+    #     }
+    #   ]
+    # }
+    schemaP = File.dirname(__FILE__) + "/../tbd.schema.json"
+    io, surfaces = processTBD(os_model, psi_set, ioP, schemaP)
+    expect(io.nil?).to be(false)
+    expect(io.is_a?(Hash)).to be(true)
+    expect(io.empty?).to be(false)
+    expect(io.has_key?(:edges))
+    expect(surfaces.nil?).to be(false)
+    expect(surfaces.is_a?(Hash)).to be(true)
+    expect(surfaces.size).to eq(23)
+    expect(surfaces.has_key?("Office Front Wall")).to be(true)
+    expect(surfaces["Office Front Wall"].has_key?(:edges)).to be(true)
+    expect(surfaces.has_key?("Office Left Wall")).to be(true)
+    expect(surfaces["Office Left Wall"].has_key?(:edges)).to be(true)
+    expect(surfaces.has_key?("Fine Storage Right Wall")).to be(true)
+    expect(surfaces["Fine Storage Right Wall"].has_key?(:edges)).to be(true)
+
+    expect(TBD.status).to eq(TBD::ERROR)
+    expect(TBD.logs.size).to eq(6)
+    # TBD.logs.each { |log| puts log[:msg] }
+    #   'clerestory' vertex count (3 or 4)
+    #   Can't override 'regular (BETBG)' PSI set  - skipping
+    #   'Office Front Wall' KHI 'beam' mismatch
```

Comment:
Is this a mismatch or is this a missing key error?  Maybe "Can't find KHI named 'beam' referenced by 'Office Front Wall'" is a better error?

Reply:
Do you want to tell the user what you are going to do when you skip here?  Are you going to use the default KHI or just skip derating?  Same thing for missing PSI.

Reply:
Good points. I'm revising the messages to be as informative as possible. I may have to seek a balance between minimal logged messages vs online guide material (with a section/tab on what TBD does when faced with missing, incomplete or invalid inputs),